private/pkg/manifest/manifest.go

```diff
@@ -136,24 +178,26 @@ func (m *Manifest) AddEntry(path string, digest Digest) error {
 	return nil
 }
 
-// Paths returns all unique paths in the manifest, order not guaranteed. If you want to iterate the
-// paths and their digests, consider using `Range` instead.
+// Paths returns all unique paths in the manifest by insertion order.
 func (m *Manifest) Paths() []string {
-	paths := make([]string, 0, len(m.pathToDigest))
-	for path := range m.pathToDigest {
-		paths = append(paths, path)
-	}
-	return paths
+	pathsCopy := make([]string, len(m.paths))
+	copy(pathsCopy, m.paths)
+	return pathsCopy
 }
 
-// Digests returns all unique digests in the manifest, order not guaranteed. If you want to iterate
-// the paths and their digests, consider using `Range` instead.
+// Digests returns all unique digests in the manifest.
+// Order is by insertion order.
 func (m *Manifest) Digests() []Digest {
 	digests := make([]Digest, 0, len(m.digestToPaths))
 	addedDigests := make(map[string]struct{}, len(m.digestToPaths))
-	// iterating over `pathToDigest` instead of `digestToPaths` to avoid handling/returning/panic on
-	// error if string -> digest parsing fails. It shouldn't.
-	for _, digest := range m.pathToDigest {
+	// Iterating over paths to guarantee ordering.
+	for _, path := range m.paths {
+		digest, ok := m.pathToDigest[path]
+		if !ok {
+			// This should be an error in the style of the rest of the codebase but
+			// this was refactored and we didn't want to change the function signature.
+			panic(fmt.Sprintf("path %q not present in pathToDigest", path))
```

Comment:
I disagree. Asking a manifest for it's digests array is a function that should never fail nor panic. The way we're iterating over the paths forces us to check an error that should never happen, and that's why we iterated over `pathToDigest`.

Reply:
For there not to be a key here, there needs to be a bug within `AddEntry` that is on a similar level to a panic - you'd have to seriously mess something up for there to be a value in `paths` but not `pathToDigest`. Even so, yes I usually return an error in this case, but I didn't want to put even more into this refactor. This isn't why the iteration was over `pathToDigest`, it was because nothing like `paths` existed.